src/Kentico.Search/SearchService.cs

```diff
@@ -53,13 +54,13 @@ public SearchService(string[] searchIndexNames, string cultureName, string siteN
 
 
         /// <summary>
-        /// Performs full-text search.
+        /// Performs paged full-text search.
         /// </summary>
-        /// <param name="query">Text to search</param>
-        /// <param name="page">Zero-based page index</param>
-        /// <param name="pageSize">Specifies the number of results per page</param>
-        /// <param name="numberOfResults">Total number of search results</param>
-        /// <returns>Search results or null if <paramref name="query"/> is empty.</returns>
+        /// <param name="query">Text to search.</param>
+        /// <param name="page">Zero-based page index.</param>
+        /// <param name="pageSize">Specifies the number of results per page. Page size must be a positive number.</param>
+        /// <param name="numberOfResults">Total number of search results for the given query which can be retrieved from the search index.</param>
+        /// <returns>Search results for the specified page or null if <paramref name="query"/> is empty.</returns>
         public virtual IEnumerable<SearchResultItem> Search(string query, int page, int pageSize, out int numberOfResults)
```

Comment:
I've kept the "page" param exactly due to backward compatibility. If we broke it, we would have to increase the package major version (according to the semantic versioning: http://semver.org/)
I plan to refactor the search service a bit in the next major version and fix the page number parameter to on-based anyway.